### ISSUE TITLE
Add support for DS216/DS416j Armada38x [$15]

### ISSUE THREAD

**Issue by Dr-Bean** [SynoCommunity/spksrc#2033]
Current toolchain on SF is 64bit...I've entered a request for a 32-bit version, so hopefully Synology will do that.

**Comment by Diaoul** [comment]
Well, we could try switching to 64bit for docker but I bet this will hurt...

**Comment by Dr-Bean** [comment]
I believe the build env Synology supplies is still 32bit, so it may be an honest mistake.

Other than that, I don't mind if someone tested with a 64bit docker, although testing the resulting packages is a different matter.
Anyway, create a separate branch with a 64b docker config, see how it goes?

**Comment by tomdbpost** [comment]
Can someone give me an update of this? Are we still waiting on Synology to release a 32 bit toolchain of SF?  Is there a way I could help, might help testing the packages?

Thnx in advance.

**Comment by Dr-Bean** [comment]
Still waiting for a response, my guess is that Synology is/has been busy with preparations for DSM6b2.
Depending on Synology's response, we might (need to) switch to a 64b build environment.

In the mean time, you can start by reading [this](https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#i-have-a-new-nas-model-what-has-to-be-done-to-provide-packages-for-it) for some background. 
As you can see, it's not unheard of that third-party packages aren't available for some time. We've seen broken toolchains or waited for multiple months for a release.

Feel free to set up a 64 bit build environment yourself, add the toolchain in spksrc, and try and build the packages yourself. Fair warning, you will run into problems with certain packages.

**Comment by bigcheek** [comment]
Hello,

I just bought a DS416j and found that unfortunately no community packages are available :(

Is there anything we can do on our side (user's side) to help you get what you need to migrate the packages to this new NAS (open a ticket with Synology asking them to give you the sources you need for example) ?

Can this help you (Armada38x sources) ?
 http://sourceforge.net/projects/dsgpl/

I also found that the Plex package is working on the DS416J.
Does this mean that someone managed to compile the toolchains for this processor ?
https://forums.plex.tv/discussion/203335/no-plex-on-ds416j

Thanks in advance; your work is greatly appreciated.

**Comment by Dr-Bean** [comment]
My previous comment already explains everything you're asking?

**Comment by tomdbpost** [comment]
@Dr-Bean Thnx for the reply! FYI I contacted Synology about the 32 bit toolchain and got this as a response... 

Hi Tom,

Thank you for contacting Synology support.
Unfortunately there is no announced plan to support this at the moment. I apologize for your inconvenience. However, you are more than welcome to send your suggestions or feature requests to our Product Management group at:
https://www.synology.com/en-global/form/inquiry/feature

However, I'm afraid I don't have any info on timeline for when such release would be implemented.

I contacted the product management group with a feature request.

**Comment by bigcheek** [comment]
@tomdbpost
Thank you for sharing this info with us.
I've juste done the same (send a feature request for the 32 bit toolchain).

@Dr-Bean 
Does this mean that the Plex team set up a 64 bit environment in order to compile their package ?
In this case they could maybe share their efforts and add the toolchain in spksrc, no ?

**Comment by bondgb** [comment]
I got back this response from Synology:

> Thank you for your interest in our products.

> Unfortunately, we do not plan to release the 32 bits version of Armada38x toolchain as we will make all toolchain in 64 bits after DSM 6.0. I apologize for the inconvenience.

> I will however forward you request to our development teams to study the possibility to release it.

**Comment by GaetanCambier** [comment]
i don't understant why you don't compile with 64bits version of toolchains ...

**Comment by tomdbpost** [comment]
@bondgb I got back somehow the same message, but in my case they made clear that they won't release a 32 bit version of Armada38x toolchain. 

@Dr-Bean @Diaoul Since Synology won't start working on a 32 bit version of Armade38x toolchain I guess we need to start with the 64bit toolchain. I have no experience with creating packages for Synology myself, but I'm willing to learn or help testing them. What is you're opinion in this guys?


The message from Synology Development:

Dear Tom,

Thank you for your interest in our products.

Unfortunately, we won't release a 32 bits version of Armada38x toolchain as we will make all toolchain in 64 bits after DSM 6.0. I apologize for the inconvenience.

You should however be able to compile with the 64 bits of toolchain if without issues if your development environment is in 64bits (processor and OS).

Best regards,

Idriss
Synology Inc.

**Comment by pcatlin** [comment]
Does this mean we will need to wait until DSM 6.0 is officially released before the 64 bit toolchain can be used?

**Comment by tomdbpost** [comment]
@Dr-Bean @Diaoul Saw that Synology added beta toolchains for DSM 6.0. Is this what you where waiting for?
http://sourceforge.net/projects/dsgpl/files/DSM%206.0%20Beta2%20Tool%20Chains/

**Comment by GaetanCambier** [comment]
Personally, i compile on 64bit machine and all package is already compiled for armada38x ...
i don't understand why you whould use a 32bit toolchains ...
i don't understant why 
the 64bit version is fully optimised for actual processor, thus the compilation time is a few reduced

**Comment by Diaoul** [comment]
Because 64bits toolchains did not exist at the time. Feel free to contribute by making a pull request for this.

**Comment by GaetanCambier** [comment]
what is the relationship between the fact that all toolchains not in 64-bit and use a 64-bit operating system ?

**Comment by Diaoul** [comment]
There is no relationship. 64bit OS complicates things with autotools and friends. Sometimes they use lib64 (instead of lib), sometimes both for lib directory so 32bit hosts and 64bit hosts are not interchangeable. As 32bit was easier to work with we went that route. That's mainly a historical reason.

**Comment by Dr-Bean** [comment]
See #2146. #2132 should fit right in after that.

**Comment by tomdbpost** [comment]
@Dr-Bean  @Diaoul Any update on this? Really looking forward to add some of the synocommunity packages to my DS216, the last update was almost 2 weeks ago. Also, I now it takes a lot of time from you, and some others to maintain the Synocommunity. is there a way I can make a donation to support you guys, specially for this issue :)!  Thanks in advance, it's really appriciated.

**Comment by piwi82** [comment]
Yes, you can use the [Donate] button on the official website : https://synocommunity.com/
(Don't use the old links at https://github.com/SynoCommunity/spksrc#donate)

**Comment by Diaoul** [comment]
I think the best for this particular issue is to create a bounty. Check the link on the first comment (https://www.bountysource.com/issues/29478304-add-support-for-ds216-ds416j-armada38x?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github)

**Comment by Dr-Bean** [comment]
Out of curiosity, anything wrong with the old links @piwi82? I left the links there because I didn't have a better place to put them at the time, but if they are broken/unwanted, we can just as well remove them.

**Comment by piwi82** [comment]
Well, I don't know for the other ones (except Diaoul's of course) but my PayPal button could be removed as I don't contribute enough improving SynoCommunity in order to receive donations.
But as Diaoul said, bounties is the best way.

**Comment by tomdbpost** [comment]
Okay thnx will check the bounty option tomorrow, is there any update guys? or still pending on #2146 and #2132 ?

**Comment by tomdbpost** [comment]
Having trouble creating an account with Bountysource, connecting with Paypall isn't working. For now, I'll pay 20 euro's via paypall to the person who fixes this issue. Yes, I now it's a small amount but it's a start and maybe some others are willing to pay some more aswell.

**Comment by sebachu** [comment]
@tomdbpost I've had good success manually installing .spk packages compiled for Armada 375. It's just a stop gap measure I know, but it got things working.

**Comment by GaetanCambier** [comment]
a simple search in google and the armada38x pakage synology can be found ...

**Comment by tomdbpost** [comment]
@sebachu Where do I find this packages? and @GaetanCambier I have been searching in Google for ages now and found nothing. Little help with the ''simple" search will be much appreciated. Actually I'm just looking for a way to let my NAS automatically do my downloads, so i'm looking for the following packages: CouchPotato, SickBeard Custom, SABnzbd. Thnx for helping me guys!

**Comment by sebachu** [comment]
@tomdbpost here are some packages compiled for Armada375 which I've modified the INFO files to make them able to be manually installed on an Armada38x. It's not a proper fix for the issue, but it worked enough for me to get my DS416j downloading.

https://www.dropbox.com/s/98olpnpp5v2pqva/Synology%20packages%20modified%20for%20Armada%2038x.zip?dl=0

**Comment by GaetanCambier** [comment]
@tomdbpost : on my github repository since 1 month ! (and compiled package too)

**Comment by tomdbpost** [comment]
Looks like I owe you guys some money, It's working!! For the other noobs like me who download the files from Dropbox from @sebachu you still need GIT installed from the Repo of Gaetan cambier for some packages to work.

**Comment by kprml** [comment]
Thanks to @GaetanCambier & @sebachu! Does anyone have a custom Sonarr package?

**Comment by GaetanCambier** [comment]
custom sonnar ? can you explain what you want ?

**Comment by kprml** [comment]
@gaetancambier Yes, on Synocommunity, there is a package called Sonarr (Nzbdrone) that is used to download. I was wondering if anyone had a custom package designed to run on the Armada 38x like the other packages above. Thanks!

**Comment by GaetanCambier** [comment]
Strange, I forgot this package, I will look tonight ;)

**Comment by kprml** [comment]
Awesome!  Thanks again. It really is a great package. Makes everything so
much easier.

**Comment by GaetanCambier** [comment]
hello,

the package Sonarr is available on my repository for arch armada38x ;)

Sorry for the delay

**Comment by kprml** [comment]
@GaetanCambier You are amazing. Thank you so much!!!

**Comment by Dr-Bean** [comment]
Packages for Armada38x have been published on the repository. Due to caching, it might take up to two hours before packages show up.

Please open a new issue for broken packages. You may report missing packages here, I haven't checked the list.

Have fun ;)

**Comment by sebachu** [comment]
Wow thanks so much everyone, package repository has been added successfully on my DS416J. Is there some way I can show my appreciation (ie bounty or donation etc)?

**Comment by Dr-Bean** [comment]
Thanks. Donations to [SynoCommunity](https://synocommunity.com) are always appreciated, helps maintain the server infra. BountySource is another option, see the link in the first comment for that.

**Comment by tomdbpost** [comment]
@Dr-Bean Thnx!!! Just me or is SickRage missing?

**Comment by Dr-Bean** [comment]
The `SickRage` package was removed a while ago. The original project (https://github.com/SiCKRAGETV/SiCKRAGE) made changes to its install method such that it couldn't be packaged (that might have changed, I don't know).
The forked project (https://github.com/SickRage/SickRage) can be installed through the `Sickbeard-Custom` package.

**Comment by Andrroid0121** [comment]
>You may report missing packages here, I haven't checked the list.

Sonarr appears to be missing on my DS416J? Thanks so much for getting this working! I thought I was doing something wrong.

**Comment by FHomer** [comment]
Thanks GaetanCambier  and Dr-Bean for your hard work

While https://synology.cambier.org/ has Sonnar, it needs Mono as a dependency, and I'm afraid the lowly DS416J wouldn't be able to handle my desired payload of nzbget, couch potato, transmission, and sonnar/mono... plus NAS functionalities

**Comment by kprml** [comment]
I have a DS416j, and run sonarr and nzbget without issue. I also run
download station, but haven't looked into couch potato yet.

**Comment by FHomer** [comment]
Did you build mono yourself ?

**Comment by kprml** [comment]
No. I installed it from synocommunity

**Comment by Dr-Bean** [comment]
Mono and Sonarr have been published. As mentioned in my previous comment, due to caching, it might take a while for the packages to show up.

**Comment by Dr-Bean** [comment]
I'm going to close this one. I believe I've published all packages on the repo, with the exception of rutorrent (in progress, related to #2186).

Before adding a new comment here, please look at [my previous comment](https://github.com/SynoCommunity/spksrc/issues/2033#issuecomment-196208366).
